### PR TITLE
chore upgrade python

### LINKED ISSUE #9: html-structure-diff python upgrade
1. [x] find . -iname '*.py' -not -path "*/venv/*" -exec pyupgrade --py311-plus {} +
3. [x] enum changed behaviour of str()
4. [x] remove dependency on async_test
5. [x] transition to pytest - https://github.com/pytest-dev/nose2pytest
6. [x] update travis CI
7. [x] ansible playbook update
8. [ ] Optional: transition to pyproject.toml
9. [x] Optional: aiohttp update?

### PR DESCRIPTION
**Pull request by b1r3k**
fixes #9

### CODE REVIEW

File: sdiff/__init__.py [hunk 1]
```diff
@@ -1,11 +1,9 @@
-from typing import Type
-
 from .parser import parse, MdParser, ZendeskHelpMdParser  # noqa
 from .renderer import TextRenderer
 from .compare import diff_struct, diff_links  # noqa
 
 
-def diff(md1, md2, renderer=TextRenderer(), parser_cls: Type[MdParser] = MdParser):
+def diff(md1, md2, renderer=TextRenderer(), parser_cls: MdParser = MdParser):
```
Comment (mshandrovskiy) [inline sdiff/__init__.py]: why is `Type` removed here?

File: sdiff/parser.py [hunk 2]
```diff
@@ -212,7 +212,7 @@ def _remove_ltr_rtl_marks(text):
     return re.sub(r'(\u200e|\u200f)', '', text)
 
 
-def parse(text, parser_cls: Type[MdParser] = MdParser):
+def parse(text, parser_cls: type[MdParser] = MdParser):
```
Comment (mshandrovskiy) [inline sdiff/parser.py:215]: recommended to use `type` over `Type`?

File: setup.py [hunk 3]
```diff
@@ -14,7 +14,8 @@ def read(f):
 ]
```
Comment (mshandrovskiy) [inline setup.py:14]: Any need to upgrade this library? `mistune`

File: tests/test_sdiff.py [hunk 4]
```diff
@@ -13,7 +13,7 @@ def _load_fixture(*path):
 
 def _read_test_files(dirpath):
     path = Path(os.path.join('tests/fixtures', dirpath))
-    filenames = set(f.name.split('.')[0] for f in path.glob('*.md'))
+    filenames = {f.name.split('.')[0] for f in path.glob('*.md')}
```
Comment (snejku) [inline tests/test_sdiff.py:16]: changing `set`  to `dict` was intended?

Comment (b1r3k) [reply tests/test_sdiff.py:16]: this is how `pyupgrade --py36-plus` works by default. `{}` can create dict if `key: value` is used or set if `value` is used as item. TBH I prefer set() as constructor but whatever..

Comment (snejku) [reply tests/test_sdiff.py:16]: ah, pardon that's set 😅

File: sdiff/parser.py (hunk 2 above)
Comment (b1r3k) [reply sdiff/parser.py:215]: this is how pyupgrade --py36-plus works by default: tries to use typing.type which fails

Additionally typing.Type usage seems to be deprecated according to: https://docs.python.org/3.11/library/typing.html#typing

so at this point using type[My_Class] should be way to go

File: setup.py (hunk 3 above)
Comment (b1r3k) [reply setup.py:14]: we are using v1 of this library: https://github.com/lepture/mistune/tree/v1 0.8.4 is last version of v1 then there is v2 and v3 which may break compatibility - I've have not checked but:

- v1 docs says it's python 3.5+ compatible
- we have tests covering usage of mistune starting from `ParserTestCase` and it seems coverage is pretty good.

File: sdiff/__init__.py (hunk 1 above)
Comment (b1r3k) [reply sdiff/__init__.py]: my bad, using type[My_Class] should be way to go

**Pull request merged**